app/controllers/pages_controller.rb

```diff
@@ -8,6 +8,14 @@ def show
     render @page
   end
 
+  def policies
+    policy_page = params[:id]
+    raise ActionController::RoutingError, "Policy page not found" unless Gemcutter::POLICY_PAGES.include?(policy_page)
+
+    add_breadcrumb t("pages.#{policy_page}.title")
+    @markdown_content = Rails.root.join("app", "views", "pages", "#{policy_page}.md").read
```

Comment:
I'm a little bit worried that we're reading and rendering a file to the user based on an input parameter, something that could be a future HackerOne report. 

Would you perhaps consider refactoring the `POLICY_PAGES` in `config/application.rb` to be a Hash of Page ID -> path to markdown. That would eliminate the arbitrary input and also remove an N+1 when checking if the page exists.

Reply:
I followed the existing pattern where only elements of the PAGES array are valid inputs.  I did make a separate policy pages array since these should have a matching md file while the pages one would not.  If you try to pass some other value than one in the array the guard clause will raise an exception.  I don't mind exploring a different approach but I'm not sure if this can be hacked.  Maybe I'm missing something?

Reply:
alternatively, I think we could remove all the page-specific logic by doing something like https://rollen.io/blog/markdown-rails/ and having the pages rendered as the default view for the action, as is done for `#show`